python/ray/util/sgd/torch/worker_group.py

```diff
@@ -0,0 +1,480 @@
+import io
+import logging
+import time
+from datetime import timedelta
+
+import ray
+import torch
+from ray.exceptions import RayActorError
+from ray.util.sgd.torch.distributed_torch_runner import \
+    LocalDistributedRunner, DistributedTorchRunner
+from ray.util.sgd.torch.torch_runner import TorchRunner
+from ray.util.sgd.utils import check_for_failure
+
+RESIZE_COOLDOWN_S = 10
+logger = logging.getLogger(__name__)
+
+
+class BaseWorkerGroup:
+    def start_workers(self, num_workers, params, dist_params,
```

Comment:
let's add docstrings for this entire interface - specify args, return values, and a high level of what the function does.

Reply:
done